arch/arch.go

```diff
@@ -0,0 +1,209 @@
+	// metadata files
+	pkgInfo, err := getPkgInfo(info, instSize)
+	if err != nil {
+		return errors.Wrap(err, "failed to create .PKGINFO")
+	}
+	f := fakeFile{".PKGINFO", pkgInfo}
+	if err := tarFile.AddFile(f.path, f, f.Open()); err != nil {
+		return errors.Wrap(err, "failed to add .PKGINFO")
+	}
+
+	if err := addInstallScript(info, tarFile); err != nil {
+		return errors.Wrap(err, "failed to add .INSTALL")
+	}
+	if err := addMtree(tarFile); err != nil {
+		return errors.Wrap(err, "failed to add .MTREE")
+	}
+	return nil
+}
+
+func (a *Arch) addPackageFiles(info *nfpm.Info, tarFile *tarhelper.File) (int64, error) {
+	files, err := info.FilesToCopy()
+	if err != nil {
+		return 0, err
+	}
+	var instSize int64
+	for _, file := range files {
+		stat, err := os.Stat(file.Source)
+		if err != nil {
+			return 0, err
+		}
+		f, err := os.Open(file.Source)
+		if err != nil {
+			return 0, err
+		}
+		err = tarFile.AddFile(file.Destination, roundingFileInfo{stat}, f)
+		f.Close()
+		if err != nil {
+			return 0, err
+		}
+		instSize += stat.Size()
+	}
+
+	for _, folder := range info.EmptyFolders {
+		if err := tarFile.MkdirAll(folder); err != nil {
+			return 0, err
+		}
+	}
+
+	return instSize, nil
+}
+
+const pkgInfoTemplate = `pkgname = {{.Info.Name}}
+pkgdesc = {{oneline .Info.Description}}
+pkgver = {{ if .Info.Epoch}}{{ .Info.Epoch }}:{{ end }}{{.Info.Version}}-{{defaultRelease .Info.Release}}
+size = {{.InstalledSize}}
+url = {{.Info.Homepage}}
+packager = {{.Info.Maintainer}}
+{{- range .Info.Replaces}}
+replaces = {{.}}
+{{- end }}
+{{- range .Info.Provides}}
+provides = {{.}}
+{{- end }}
+{{- range .Info.Depends}}
+depend = {{.}}
+{{- end }}
+{{- range .Info.Recommends}}
+optdepend = {{.}}
+{{- end }}
+{{- range .Info.Suggests}}
+optdepend = {{.}}
+{{- end }}
+{{- range .Info.Conflicts}}
+conflict = {{.}}
+{{- end }}
+{{- range .Info.ArchLinux.Groups}}
+group = {{.}}
+{{- end }}
+{{- range .Info.ConfigFiles}}
+backup = {{.}}
+{{- end}}
+`
+
+type controlData struct {
+	Info          *nfpm.Info
+	InstalledSize int64
+}
+
+func getPkgInfo(info *nfpm.Info, instSize int64) ([]byte, error) {
+	buf := &bytes.Buffer{}
+	tmpl := template.New("pkginfo")
+	tmpl.Funcs(template.FuncMap{
+		// the description field only supports single-line descriptions
+		"oneline": func(str string) string {
+			return strings.TrimRight(regexp.MustCompile(`\n\s*`).ReplaceAllString(str, " "), " \n")
+		},
+		// the full version must have a hyphen followed by the release; the release
+		// can be empty (e.g., a version of "1.2.3-"), but that looks weird
+		"defaultRelease": func(str string) string {
+			if str == "" {
+				return "1"
+			}
+			return str
+		},
+	})
+	if err := template.Must(tmpl.Parse(pkgInfoTemplate)).Execute(buf, controlData{info, instSize}); err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+func addInstallScript(info *nfpm.Info, tarFile *tarhelper.File) error {
+	if info.ArchLinux.InstallScript == "" {
```

Comment:
we'll need to add to the docs that the `scripts` field is not effective on arch packages... 

or maybe we can use one field from there? not sure what's less confusing 🤔